### PR TITLE
Docs: Add password keystore setting for email account passwords

### PR DESCRIPTION
**Pull request by spinscale**
The keystore configuration options have not been mentioned yet, but have so in hipchat/jira/pagerduty actions.

### CODE REVIEW

File: x-pack/docs/en/watcher/actions/email.asciidoc [hunk 1]
```diff
@@ -349,9 +357,18 @@ xpack.notification.email.account:
             host: smtp-mail.outlook.com
             port: 587
             user: <email.address>
-            password: <password>
 --------------------------------------------------
 
+In order to not store the account SMTP password as plaintext in the
+elasticsearch yaml file, you should use the keystore instead (you can still
+store the password in the elasticsearch configuration file, if you need to)
```
Comment (rjernst) [inline x-pack/docs/en/watcher/actions/email.asciidoc]: I don't think we should have this node about putting the password in elasticsearch.yml. This should be going away; we shouldn't be promoting it.

File: x-pack/docs/en/watcher/actions/email.asciidoc [hunk 2]
```diff
@@ -349,9 +357,18 @@ xpack.notification.email.account:
             host: smtp-mail.outlook.com
             port: 587
             user: <email.address>
-            password: <password>
 --------------------------------------------------
 
+In order to not store the account SMTP password as plaintext in the
+elasticsearch yaml file, you should use the keystore instead (you can still
```
Comment (rjernst) [inline x-pack/docs/en/watcher/actions/email.asciidoc]: We should crosslink to the keystore docs ("secure settings").

**Review by rjernst — COMMENTED** [review]
A couple comments.

File: x-pack/docs/en/watcher/actions/email.asciidoc (hunk 1 above)
Comment (spinscale) [reply x-pack/docs/en/watcher/actions/email.asciidoc]: removed that mention

File: x-pack/docs/en/watcher/actions/email.asciidoc (hunk 2 above)
Comment (spinscale) [reply x-pack/docs/en/watcher/actions/email.asciidoc]: done

**Review by danielmitterdorfer — APPROVED** [review]
LGTM

**Pull request merged**